diff -ru a/usr/sbin/init-beagle-flasher b/usr/sbin/init-beagle-flasher
--- a/usr/sbin/init-beagle-flasher	2023-02-22 15:12:47.646992674 +0100
+++ b/usr/sbin/init-beagle-flasher	2023-02-22 14:54:28.920366351 +0100
@@ -375,6 +375,11 @@
 		reset_leds
 	fi
 
+	# Render SD card unbootable to prevent a reboot into the flasher
+	mount -o remount,rw /
+	mv /boot/uEnv.txt /boot/uEnv.txt.orig
+	mount -o remount,ro /
+
 	unset are_we_flasher
 	are_we_flasher=$(grep init-beagle-flasher /proc/cmdline || true)
 	if [ ! "x${are_we_flasher}" = "x" ] ; then
@@ -494,4 +499,17 @@
 fi
 
 check_running_system
+
+# Copy sensor configuration to root partition
+mount -o remount,rw /
+mount ${source}p1 /boot/uboot
+find /boot/uboot -maxdepth 1 -name '*.tar.gz' -exec cp {} /opt/config.tar.gz \;
+flush_cache
+umount /boot/uboot
+if [ ! -f /opt/config.tar.gz ]; then
+  echo "Error: No sensor configuration found, aborting."
+  exit 1
+fi
+mount -o remount,ro /
+
 partition_drive
diff -ru a/usr/sbin/init-shutdown-beagle-flasher b/usr/sbin/init-shutdown-beagle-flasher
--- a/usr/sbin/init-shutdown-beagle-flasher	2023-02-21 13:37:06.226439584 +0100
+++ b/usr/sbin/init-shutdown-beagle-flasher	2023-02-21 13:36:12.477490572 +0100
@@ -5,13 +5,13 @@
 unset are_we_flasher
 are_we_flasher=$(grep init-beagle-flasher-test /proc/cmdline || true)
 if [ ! "x${are_we_flasher}" = "x" ] ; then
-	systemctl poweroff || halt
+	systemctl reboot || reboot
 	exit
 fi
 
 unset are_we_flasher
 are_we_flasher=$(grep init-beagle-flasher /proc/cmdline || true)
 if [ ! "x${are_we_flasher}" = "x" ] ; then
-	systemctl poweroff || halt
+	systemctl reboot || reboot
 	exit
 fi
