diff -ur a/usr/sbin/init-beagle-flasher b/usr/sbin/init-beagle-flasher
--- a/usr/sbin/init-beagle-flasher	2025-03-14 13:07:52.144000897 +0000
+++ b/usr/sbin/init-beagle-flasher	2025-03-14 13:23:47.968001011 +0000
@@ -371,6 +371,11 @@
 		reset_leds
 	fi
 
+	# Render SD card unbootable to prevent a reboot into the flasher
+	mount -o remount,rw /
+	mv /boot/uEnv.txt /boot/uEnv.txt.orig
+	mount -o remount,ro /
+
 	unset are_we_flasher
 	are_we_flasher=$(grep init-beagle-flasher /proc/cmdline || true)
 	if [ ! "x${are_we_flasher}" = "x" ] ; then
@@ -393,8 +398,8 @@
 			message="--------------------------------------------------------------------------------" ; broadcast
 			#For sync
 			echo s > /proc/sysrq-trigger
-			#For shutdown the system
-			echo o > /proc/sysrq-trigger
+			#Reboot the system
+			echo b > /proc/sysrq-trigger
 		fi
 
 		message="4"                                                                                ; broadcast
@@ -519,4 +524,17 @@
 fi
 
 check_running_system
+
+# Copy sensor configuration to root partition
+mount -o remount,rw /
+mount ${source}p1 /boot/uboot
+find /boot/uboot -maxdepth 1 -name '*.tar.gz' -exec cp {} /opt/config.tar.gz \;
+flush_cache
+umount /boot/uboot
+if [ ! -f /opt/config.tar.gz ]; then
+  echo "Error: No sensor configuration found, aborting."
+  exit 1
+fi
+mount -o remount,ro /
+
 partition_drive
